sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java

```diff
@@ -1010,6 +1037,118 @@ public boolean getValidate() {
     private Write() {}
   }
 
+  /**
+   * {@link BigQuerySink} is implemented as a {@link FileBasedSink}.
+   *
+   * <p>It uses BigQuery load job to import files into BigQuery.
+   */
+  static class BigQuerySink<T> extends FileBasedSink<T> {
+    private final String jobIdToken;
+    private final String jsonTable;
+    private final String jsonSchema;
+    private final WriteDisposition writeDisposition;
+    private final CreateDisposition createDisposition;
+    private final Coder<T> coder;
+
+    public BigQuerySink(
+        String jobIdToken,
+        String jsonTable,
+        String jsonSchema,
+        WriteDisposition writeDisposition,
+        CreateDisposition createDisposition,
+        String tempFile,
+        Coder<T> coder) {
+      super(tempFile, "");
+      this.jobIdToken = jobIdToken;
+      this.jsonTable = jsonTable;
+      this.jsonSchema = jsonSchema;
+      this.writeDisposition = writeDisposition;
+      this.createDisposition = createDisposition;
+      this.coder = coder;
+    }
+
+    @Override
+    public FileBasedSink.FileBasedWriteOperation<T> createWriteOperation(PipelineOptions options) {
+      return new BigQueryWriteOperation<>(
+          this, jobIdToken, jsonTable, jsonSchema, writeDisposition, createDisposition, coder);
+    }
+
+    private static class BigQueryWriteOperation<T> extends FileBasedWriteOperation<T> {
+      private final String jobIdToken;
+      private final String jsonTable;
+      private final String jsonSchema;
+      private final WriteDisposition writeDisposition;
+      private final CreateDisposition createDisposition;
+      private final Coder<T> coder;
+
+      private BigQueryWriteOperation(
+          BigQuerySink<T> sink,
+          String jobIdToken,
+          String jsonTable,
+          String jsonSchema,
+          WriteDisposition writeDisposition,
+          CreateDisposition createDisposition,
+          Coder<T> coder) {
```

Comment:
aren't all these parameters on the `sink`? We should not need to pass them in addition, just use getters instead.